Web search doesn't work (null error)

### Problem
Enabled Web Search feature, enabled Google Web Search API Key, created and added my key, created and added my Google Custom Search Engine ID. Clicked the toggle button on the chat frame to turn web search ON. Typed "test" and sent request.

Error notification comes up with message: `An unexpected error occurred: ExecutionException - Error in web search prompt execution Caused by: NullPointerException - Cannot invoke "dev.langchain4j.data.message.UserMessage.singleText()" because the return value of "com.devoxx.genie.model.request.ChatMessageContext.getUserMessage()" is null Please check the IDE log for more details.`

Note: non-web search queries work as expected.

### Environment
IntelliJ IDEA 2025.2.4 (Ultimate Edition)
Build #IU-252.27397.103, built on October 23, 2025

C:\Users\xxxxxxxx>java -version
java version "25.0.1" 2025-10-21 LTS
Java(TM) SE Runtime Environment (build 25.0.1+8-LTS-27)
Java HotSpot(TM) 64-Bit Server VM (build 25.0.1+8-LTS-27, mixed mode, sharing)

DevoxxGenie version 0.7.0
Ollama LLM (devstral)

### Partial Error Log
```
2025-10-30 23:36:03,476 [  15256]   INFO - STDOUT - 23:36:03.476 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewJavaScriptExecutor] [23:36:03.476][DEBUG][#10] JavaScript execution #{}: {}
2025-10-30 23:36:03,499 [  15279]   INFO - STDOUT - 23:36:03.499 [AWT-EventQueue-0] INFO  c.d.g.u.w.h.WebViewAIMessageUpdater - Executing JavaScript to add user message
2025-10-30 23:36:03,499 [  15279]   INFO - STDOUT - 23:36:03.499 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewJavaScriptExecutor] [23:36:03.499][DEBUG][#11] JavaScript execution #{}: {}
2025-10-30 23:36:03,511 [  15291]   INFO - STDOUT - 23:36:03.509 [AWT-EventQueue-0] ERROR c.d.g.s.p.s.AbstractPromptExecutionStrategy - Error in web search prompt execution: Cannot invoke "dev.langchain4j.data.message.UserMessage.singleText()" because the return value of "com.devoxx.genie.model.request.ChatMessageContext.getUserMessage()" is null
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - java.lang.NullPointerException: Cannot invoke "dev.langchain4j.data.message.UserMessage.singleText()" because the return value of "com.devoxx.genie.model.request.ChatMessageContext.getUserMessage()" is null
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.strategy.WebSearchPromptStrategy.executeStrategySpecific(WebSearchPromptStrategy.java:35)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.strategy.AbstractPromptExecutionStrategy.execute(AbstractPromptExecutionStrategy.java:94)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.PromptExecutionService.executePrompt(PromptExecutionService.java:99)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.devoxx.genie.controller.PromptExecutionController.executePromptWithContext(PromptExecutionController.java:92)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.devoxx.genie.controller.PromptExecutionController.lambda$handlePromptSubmission$0(PromptExecutionController.java:74)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.base/java.util.Optional.ifPresentOrElse(Optional.java:196)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.devoxx.genie.controller.PromptExecutionController.handlePromptSubmission(PromptExecutionController.java:73)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.devoxx.genie.controller.ActionButtonsPanelController.handlePromptSubmission(ActionButtonsPanelController.java:91)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.devoxx.genie.ui.panel.ActionButtonsPanel.onSubmitPrompt(ActionButtonsPanel.java:210)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.devoxx.genie.ui.panel.ActionButtonsPanel.lambda$onPromptSubmitted$6(ActionButtonsPanel.java:289)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:240)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:26)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:202)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.AppImplKt$runnableUnitFunction$1.invoke(appImpl.kt:104)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.AppImplKt$runnableUnitFunction$1.invoke(appImpl.kt:104)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.doRunWriteIntentReadAction(NestedLocksThreadingSupport.kt:666)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.runPreventiveWriteIntentReadAction(NestedLocksThreadingSupport.kt:640)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.runWriteIntentReadAction(NestedLocksThreadingSupport.kt:633)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:1022)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.ApplicationImpl$6.run(ApplicationImpl.java:574)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:167)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:167)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:173)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:167)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:27)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:122)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:43)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:728)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:750)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:595)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:488)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12$lambda$11$lambda$10$lambda$9(IdeEventQueue.kt:313)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:865)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12$lambda$11$lambda$10(IdeEventQueue.kt:312)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$3(IdeEventQueue.kt:974)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:110)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:974)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12(IdeEventQueue.kt:307)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:347)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
2025-10-30 23:36:03,512 [  15292]   INFO - STDOUT - 23:36:03.512 [AWT-EventQueue-0] ERROR c.d.g.s.prompt.error.PromptException - ERROR:Error in web search prompt execution - false
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 23:36:03.513 [AWT-EventQueue-0] ERROR c.d.g.s.p.error.PromptErrorHandler - Error in web search prompt execution
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - java.lang.NullPointerException: Cannot invoke "dev.langchain4j.data.message.UserMessage.singleText()" because the return value of "com.devoxx.genie.model.request.ChatMessageContext.getUserMessage()" is null
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.strategy.WebSearchPromptStrategy.executeStrategySpecific(WebSearchPromptStrategy.java:35)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.strategy.AbstractPromptExecutionStrategy.execute(AbstractPromptExecutionStrategy.java:94)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.devoxx.genie.service.prompt.PromptExecutionService.executePrompt(PromptExecutionService.java:99)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.devoxx.genie.controller.PromptExecutionController.executePromptWithContext(PromptExecutionController.java:92)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.devoxx.genie.controller.PromptExecutionController.lambda$handlePromptSubmission$0(PromptExecutionController.java:74)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.base/java.util.Optional.ifPresentOrElse(Optional.java:196)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.devoxx.genie.controller.PromptExecutionController.handlePromptSubmission(PromptExecutionController.java:73)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.devoxx.genie.controller.ActionButtonsPanelController.handlePromptSubmission(ActionButtonsPanelController.java:91)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.devoxx.genie.ui.panel.ActionButtonsPanel.onSubmitPrompt(ActionButtonsPanel.java:210)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.devoxx.genie.ui.panel.ActionButtonsPanel.lambda$onPromptSubmitted$6(ActionButtonsPanel.java:289)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:240)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:26)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:202)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.AppImplKt$runnableUnitFunction$1.invoke(appImpl.kt:104)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.AppImplKt$runnableUnitFunction$1.invoke(appImpl.kt:104)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.doRunWriteIntentReadAction(NestedLocksThreadingSupport.kt:666)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.runPreventiveWriteIntentReadAction(NestedLocksThreadingSupport.kt:640)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.runWriteIntentReadAction(NestedLocksThreadingSupport.kt:633)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:1022)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.ApplicationImpl$6.run(ApplicationImpl.java:574)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:167)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:167)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:173)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:167)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:27)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:122)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:43)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:728)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:750)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:595)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:488)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12$lambda$11$lambda$10$lambda$9(IdeEventQueue.kt:313)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:865)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12$lambda$11$lambda$10(IdeEventQueue.kt:312)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$3(IdeEventQueue.kt:974)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:110)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:974)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$12(IdeEventQueue.kt:307)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:347)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
2025-10-30 23:36:03,514 [  15294]   INFO - STDOUT - 	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
2025-10-30 23:36:03,529 [  15309]   INFO - STDOUT - 23:36:03.528 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewJavaScriptExecutor] TIMING[jsExecution#7]: 0ms
2025-10-30 23:36:03,529 [  15309]   INFO - STDOUT - 23:36:03.529 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewJavaScriptExecutor] TIMING[jsExecution#8]: 0ms
2025-10-30 23:36:03,610 [  15390]   INFO - STDOUT - 23:36:03.610 [nioEventLoopGroup-3-2] INFO  c.devoxx.genie.ui.webview.WebServer - Handling request for: /icons/copy.svg
2025-10-30 23:36:03,610 [  15390]   INFO - STDOUT - 23:36:03.610 [nioEventLoopGroup-3-2] INFO  c.devoxx.genie.ui.webview.WebServer - Serving content with type: image/svg+xml
2025-10-30 23:36:03,642 [  15422]   INFO - STDOUT - 23:36:03.642 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewJavaScriptExecutor] [23:36:03.642][DEBUG][#12] JavaScript execution #{}: {}
2025-10-30 23:36:03,716 [  15496]   INFO - STDOUT - 23:36:03.716 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewSleepWakeRecoveryHandler] [23:36:03.716][DEBUG][#52] Component resized - potential wake event
2025-10-30 23:36:03,717 [  15497]   INFO - STDOUT - 23:36:03.716 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewSleepWakeRecoveryHandler] [23:36:03.716][DEBUG][#53] Component event passed debounce check - proceeding with wake event check: {}
2025-10-30 23:36:03,717 [  15497]   INFO - STDOUT - 23:36:03.716 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewSleepWakeRecoveryHandler] [23:36:03.716][DEBUG][#54] Checking for potential wake event - reason: {}
2025-10-30 23:36:03,717 [  15497]   INFO - STDOUT - 23:36:03.716 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewSleepWakeRecoveryHandler] [23:36:03.716][DEBUG][#55] Evaluating if conservative recovery should be attempted
2025-10-30 23:36:03,717 [  15497]   INFO - STDOUT - 23:36:03.717 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewSleepWakeRecoveryHandler] [23:36:03.717][DEBUG][#56] Component has black background while showing - potential black rectangle
2025-10-30 23:36:03,717 [  15497]   INFO - STDOUT - 23:36:03.717 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewSleepWakeRecoveryHandler] [23:36:03.717][DEBUG][#57] Conservative recovery evaluation - browserStateIssue: {}, renderingIssue: {}, strongIndicators: {}, shouldRecover: {}
2025-10-30 23:36:03,717 [  15497]   INFO - STDOUT - 23:36:03.717 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewSleepWakeRecoveryHandler] [23:36:03.717][DEBUG][#58] Recovery determined to be needed based on actual detected issues
2025-10-30 23:36:03,717 [  15497]   INFO - STDOUT - 23:36:03.717 [AWT-EventQueue-0] INFO  c.d.g.u.w.handler.WebViewDebugLogger - [WebViewSleepWakeRecoveryHandler] [23:36:03.717][INFO][#59] Scheduling recovery - reason: {}
2025-10-30 23:36:03,717 [  15497]   INFO - STDOUT - 23:36:03.717 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewSleepWakeRecoveryHandler] [23:36:03.717][DEBUG][#60] Cancelled existing recovery timer
2025-10-30 23:36:03,717 [  15497]   INFO - STDOUT - 23:36:03.717 [AWT-EventQueue-0] INFO  c.d.g.u.w.handler.WebViewDebugLogger - [WebViewSleepWakeRecoveryHandler] [23:36:03.717][INFO][#61] Recovery scheduled in 3 seconds (reason: {})
2025-10-30 23:36:03,733 [  15513]   INFO - STDOUT - 23:36:03.733 [AWT-EventQueue-0] DEBUG c.d.g.u.w.handler.WebViewDebugLogger - [WebViewJavaScriptExecutor] TIMING[jsExecution#9]: 2ms
```

--- a/src/main/java/com/devoxx/genie/service/prompt/strategy/WebSearchPromptStrategy.java
+++ b/src/main/java/com/devoxx/genie/service/prompt/strategy/WebSearchPromptStrategy.java
@@ -32,7 +32,7 @@ public class WebSearchPromptStrategy extends AbstractPromptExecutionStrategy {
             @NotNull PromptOutputPanel panel,
             @NotNull PromptTask<PromptResult> resultTask) {
         
-        log.debug("Web search with query: {}", context.getUserMessage().singleText());
+        log.debug("Web search with query: {}", context.getUserPrompt());
 
         // Add user message to context and memory
         // prepareMemory(context);

